mysql-test/suite/tianmu/t/auto_increment.test

```diff
@@ -0,0 +1,154 @@
+--disable_warnings
+drop table if exists tint;
+drop table if exists ttinyint;
+drop table if exists tsmallint;
+drop table if exists tmediumint;
+drop table if exists tbigint;
+--enable_warnings
+CREATE TABLE tint (
+  id int(4) NOT NULL AUTO_INCREMENT,
+  data VARCHAR(64) DEFAULT NULL,
+  PRIMARY KEY (id)
+)  engine=tianmu;
+
+insert into tint (id, data) values(2147483646, "first"),(63, "middle"),(0, "last");
+select * from tint;
+--error 1062
+insert into tint (id, data) values (0, "last");
+select * from tint;
+
+CREATE TABLE ttinyint (
+  id tinyint(1) NOT NULL AUTO_INCREMENT,
+  data VARCHAR(64) DEFAULT NULL,
+  PRIMARY KEY (id)
+)  engine=tianmu;
+
+insert into ttinyint (id, data) values(126, "first"),(63, "middle"),(0, "last");
+select * from ttinyint;
+--error 1062
+insert into ttinyint (id, data) values (0, "last");
+select * from ttinyint;
+
+CREATE TABLE tsmallint (
+  id smallint(2) NOT NULL AUTO_INCREMENT,
+  data VARCHAR(64) DEFAULT NULL,
+  PRIMARY KEY (id)
+)  engine=tianmu;
+
+insert into tsmallint (id, data) values(32766, "first"),(63, "middle"),(0, "last");
+select * from tsmallint;
+--error 1062
+insert into tsmallint (id, data) values (0, "last");
+select * from tsmallint;
+
+CREATE TABLE tmediumint (
+  id mediumint(3) NOT NULL AUTO_INCREMENT,
+  data VARCHAR(64) DEFAULT NULL,
+  PRIMARY KEY (id)
+)  engine=tianmu;
+
+insert into tmediumint (id, data) values(8388606, "first"),(63, "middle"),(0, "last");
+select * from tmediumint;
+--error 1062
+insert into tmediumint (id, data) values (0, "last");
+select * from tmediumint;
+
+CREATE TABLE tbigint (
+  id bigint(8) NOT NULL AUTO_INCREMENT,
+  data VARCHAR(64) DEFAULT NULL,
+  PRIMARY KEY (id)
+)  engine=tianmu;
+
+insert into tbigint (id, data) values(9223372036854775806, "first"),(63, "middle"),(0, "last");
+select * from tbigint;
+--error 1062
+insert into tbigint (id, data) values (0, "last");
+select * from tbigint;
+
+# unsigned data type
+
+--disable_warnings
+drop table if exists tint;
+drop table if exists ttinyint;
+drop table if exists tsmallint;
+drop table if exists tmediumint;
+drop table if exists tbigint;
+--enable_warnings
+CREATE TABLE tint (
+  id int(4) unsigned NOT NULL AUTO_INCREMENT,
+  data VARCHAR(64) DEFAULT NULL,
+  PRIMARY KEY (id)
+)  engine=tianmu;
+
+insert into tint (id, data) values(4294967294, "first"),(63, "middle"),(0, "last");
+select * from tint;
+--error 1062
+insert into tint (id, data) values (0, "last");
+select * from tint;
+
+CREATE TABLE ttinyint (
+  id tinyint(1) unsigned NOT NULL AUTO_INCREMENT,
+  data VARCHAR(64) DEFAULT NULL,
+  PRIMARY KEY (id)
+)  engine=tianmu;
+
+insert into ttinyint (id, data) values(254, "first"),(63, "middle"),(0, "last");
+select * from ttinyint;
+--error 1062
+insert into ttinyint (id, data) values (0, "last");
+select * from ttinyint;
+
+CREATE TABLE tsmallint (
+  id smallint(2) unsigned NOT NULL AUTO_INCREMENT,
+  data VARCHAR(64) DEFAULT NULL,
+  PRIMARY KEY (id)
+)  engine=tianmu;
+
```

Comment:
Notice that
`
When you insert any other value into an AUTO_INCREMENT column, the column is set to that value and the sequence is reset so that the next automatically generated value follows sequentially from the largest column value. For example:
`

Reply:
Updating an existing AUTO_INCREMENT column value also resets the AUTO_INCREMENT sequence.

Reply:
To start with an AUTO_INCREMENT value other than 1, set that value with [CREATE TABLE](https://dev.mysql.com/doc/refman/8.0/en/create-table.html) or [ALTER TABLE](https://dev.mysql.com/doc/refman/8.0/en/alter-table.html), like this:

For more infor, pls refer to :
https://dev.mysql.com/doc/refman/8.0/en/example-auto-increment.html

Reply:
These features of the auto increment will be added in the issue #1072 and #1059